### PR TITLE
#939: flushing onChangeDebounced when blurring or submitting

### PR DESCRIPTION
**Pull request by slaude**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [ ] One core team UX approval

### CODE REVIEW

File: src/components/TextField/TextField.spec.jsx [hunk 1]
```diff
@@ -143,4 +143,69 @@ describe('TextField', () => {
 
 		assert(event.persist.called);
 	});
+
+	it('should flush the `onChangeDebounced`-handled event before blurring', () => {
+		const onBlur = sinon.spy();
+		const internalChangeSpy = sinon.spy();
+		const onChangeDebounced = () => {
+			assert(onBlur.notCalled);
+			internalChangeSpy();
```
Comment (slaude) [inline src/components/TextField/TextField.spec.jsx]: this seems like a hack, so if someone wants to suggest a way to ensure the order of operations in a more clean way, i'm all 👂

File: src/components/TextField/TextField.jsx [hunk 2]
```diff
@@ -176,15 +176,18 @@ const TextField = createClass({
 	},
 
 	handleBlur(event) {
-		const { onBlur } = this.props;
+		const { onBlur, onChangeDebounced } = this.props;
 
 		const value = _.get(event, 'target.value', '');
 
+		if (onChangeDebounced !== _.noop) {
+			this._handleChangeDebounced.flush();
```
Comment (tarun) [inline src/components/TextField/TextField.jsx]: how about just calling flush always - instead of checking against `lodash.noop`...

**Comment by jondlm** [comment]
I'm torn between a patch and a minor...

**Comment by slaude** [comment]
fixes #939

**Pull request merged**